explorer/interpreter/type_checker.cpp

```diff
@@ -2123,6 +2123,20 @@ auto TypeChecker::TypeCheckExp(Nonnull<Expression*> e,
           op.set_rewritten_form(*converted);
           return Success();
         }
+        case Operator::NotEq: {
+          ErrorOr<Nonnull<Expression*>> converted = BuildBuiltinMethodCall(
+              impl_scope, op.arguments()[0],
+              BuiltinInterfaceName{Builtins::EqWith, ts[1]},
+              BuiltinMethodCall{"NotEqual", op.arguments()[1]});
+          if (!converted.ok()) {
+            // We couldn't find a matching `impl`.
+            return CompilationError(e->source_loc())
+                   << *ts[0] << " is not equal comparable with " << *ts[1]
+                   << " (" << converted.error().message() << ")";
+          }
+          op.set_rewritten_form(*converted);
+          return Success();
+        }
```

Comment:
Can you reduce the duplication here? The only difference between these two cases is (should be) the `"Equal"` versus `"NotEqual"`, so perhaps we could have a single implementation with an `op.op() == Operator::Eq ? "Equal" : "NotEqual"`, or perhaps the implementation could be factored out into a lambda to which the right string is passed?

Reply:
Sure

Reply:
DONE!